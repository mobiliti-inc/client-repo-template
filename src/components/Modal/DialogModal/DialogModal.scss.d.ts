// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'box-container': string;
  'button-group': string;
  'cancel-button-container': string;
  'fades-in-out': string;
  'main-container': string;
  'save-button-container': string;
  'save-button-container--disabled': string;
  'title': string;
}
declare var cssExports: CssExports;
export = cssExports;
