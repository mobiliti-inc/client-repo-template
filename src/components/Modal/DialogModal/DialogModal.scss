@import "../../../styles/_variables.scss";

.main-container {
  display: flex;
  flex-direction: column;
  width: 80%;
}

.title {
  max-height: 62px;
  max-width: 286px;
  color: $darker-blue;
  font-family: $futura;
  font-size: 22px;
  font-weight: bold;
  letter-spacing: -0.49px;
  line-height: 31px;
  margin: 0 0 20px 0;
  word-wrap: break-word;
}

.box-container {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.button-group {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 8px;
}

.save-button-container {
  width: 80%;
  opacity: 1;

  &--disabled {
    opacity: 0.6;
  }
}

.cancel-button-container {
  align-self: center;
  height: 25px;
  width: 124px;
  font-family: $futura;
  font-size: 14px;
  font-weight: bold;
  line-height: 23px;
  text-align: center;

  & button {
    color: $blue;
    border: none;
    background-color: white;
    cursor: pointer;

    &:focus {
      outline: none;
    }
  }
}