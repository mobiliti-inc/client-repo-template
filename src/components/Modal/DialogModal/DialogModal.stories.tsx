import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import * as React from 'react';
import { DialogModal } from '../../';

storiesOf('DialogModal', module)
	.add('default', () =>
		<DialogModal
			blueButtonClick={() => { }}
			blueButtonText={text('blueButtonText', "Continue")}
			closeModal={() => { }}
			footer={boolean('footer', true)}
			isBlueButtonDisabled={boolean('isBlueButtonDisabled', false)}
			modalIsVisible={boolean('modalIsVisible', true)}
			modalTitle={text('modalTitle', "Modal title")}
			whiteButtonClick={() => { }}
			whiteButtonText={text('whiteButtonText', "Back")}
		>
			<h2>Modal Body</h2>
		</DialogModal>
	);
