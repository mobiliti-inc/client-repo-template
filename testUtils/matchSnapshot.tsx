const { shallowToJson } = require('enzyme-to-json');

export const matchSnapshot = (subject: any) => {
	expect(shallowToJson(subject)).toMatchSnapshot();
};
