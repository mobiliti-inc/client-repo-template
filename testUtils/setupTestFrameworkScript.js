const React = require("react");
// const CSSModules = require("react-css-modules");
const Enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

const { shallow, render, mount } = Enzyme;

Enzyme.configure({ adapter: new Adapter() });
jest.setTimeout(30000);

global.React = React;
// global.CSSModules = CSSModules;
global.shallow = shallow;
global.render = render;
global.mount = mount;
