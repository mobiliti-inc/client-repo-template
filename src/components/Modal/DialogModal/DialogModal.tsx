import React from 'react';
// @ts-ignore
import Modal, { EasyModalHeader } from 'easy-modal-react';

import { Button } from '../../';

import './DialogModal.scss';

interface DialogModalProps {
	blueButtonClick?: (...args: any[]) => any;
	whiteButtonClick?: (...args: any[]) => any;
	modalIsVisible?: boolean;
	closeModal: (...args: any[]) => any;
	modalTitle: string;
	whiteButtonText?: string;
	blueButtonText?: string;
	isBlueButtonDisabled?: boolean;
	className?: string;
	footer?: boolean;
}

const DialogModal: React.FC<DialogModalProps> = props => {
	const {
		blueButtonClick,
		modalTitle,
		isBlueButtonDisabled,
		blueButtonText,
		whiteButtonText,
		children,
		whiteButtonClick,
		modalIsVisible,
		closeModal,
		footer
	} = props;

	return (
		<Modal
			open={modalIsVisible}
			onClose={closeModal}
			header={<EasyModalHeader onClose={closeModal} />}
			footer={
				footer ? (
					<div styleName="button-group">
						{blueButtonText && (
							<div
								styleName={`save-button-container ${isBlueButtonDisabled ? 'save-button-container--disabled' : ''}`}
							>
								<Button onClick={blueButtonClick} disabled={isBlueButtonDisabled}>
									{blueButtonText}
								</Button>
							</div>
						)}
						{whiteButtonText && (
							<div styleName="cancel-button-container">
								<button onClick={whiteButtonClick}>{whiteButtonText}</button>
							</div>
						)}
					</div>
				) : null
			}
		>
			<div styleName="title">{modalTitle}</div>
			{children}
		</Modal>
	);
};

export default DialogModal;
