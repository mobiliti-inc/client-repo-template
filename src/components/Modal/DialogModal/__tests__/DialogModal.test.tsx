import React from 'react';
import { shallow } from 'enzyme';

import { matchSnapshot } from '../../../../../testUtils/matchSnapshot';

import DialogModal from '../DialogModal';

describe('DialogModal', () => {
	it('renders', () => {
		const tree = shallow(
			<DialogModal
				modalTitle="test headline"
				blueButtonClick={() => { }}
				whiteButtonClick={() => { }}
				modalIsVisible
				closeModal={() => { }}
				whiteButtonText="test"
				blueButtonText="message test"
				isBlueButtonDisabled
			/>
		);
		matchSnapshot(tree);
	});

	it('renders with children', () => {
		const tree = shallow(
			<DialogModal
				modalTitle="test headline"
				blueButtonClick={() => { }}
				whiteButtonClick={() => { }}
				modalIsVisible
				closeModal={() => { }}
				whiteButtonText="test"
				blueButtonText="message test"
				isBlueButtonDisabled={false}
			>
				<div />
			</DialogModal>
		);
		matchSnapshot(tree);
	});
});
