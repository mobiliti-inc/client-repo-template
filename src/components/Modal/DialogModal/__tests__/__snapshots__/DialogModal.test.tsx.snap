// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogModal renders 1`] = `
<t
  className=""
  footer={null}
  header={
    <t
      onClose={[Function]}
    />
  }
  onClose={[Function]}
  open={true}
>
  <div
    className="title"
  >
    test headline
  </div>
</t>
`;

exports[`DialogModal renders with children 1`] = `
<t
  className=""
  footer={null}
  header={
    <t
      onClose={[Function]}
    />
  }
  onClose={[Function]}
  open={true}
>
  <div
    className="title"
  >
    test headline
  </div>
  <div />
</t>
`;
